pkg/reconciler/common/transformer_inject_envvar.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2025 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"os"
+
+	mf "github.com/manifestival/manifestival"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	KUBERNETES_MIN_VERSION_ENV_KEY = "KUBERNETES_MIN_VERSION"
+)
+
+func DeploymentEnvVarKubernetesMinVersion() mf.Transformer {
+	var envVars []corev1.EnvVar
+	if minVersion, exists := os.LookupEnv(KUBERNETES_MIN_VERSION_ENV_KEY); exists {
+		envVars = append(envVars, corev1.EnvVar{
+			Name:  KUBERNETES_MIN_VERSION_ENV_KEY,
+			Value: minVersion,
+		})
+	}
+	return DeploymentEnvVars(envVars)
+}
+
+func DeploymentEnvVars(envVars []corev1.EnvVar) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		if u.GetKind() != "Deployment" {
+			return nil
+		}
```

Comment:
Since the remote-resolvers and pipelines-controller both can run as StatefulSets, do we need to support `u.GetKind() == "StatefulSet"` here or would that belong in a separate function like `StatefulSetEnvVars()`?

Reply:
Well, in. theory, yes, but until now we assumed all components are deployd as deployment. Statefulsets are kind of transformers that comes at the end of all other transformations, to adress your point we may need to review all the transformers functions which means adding a bench of tests unit and e2e tests, but probably it is useless for now.